### DISCUSSION TITLE
Is there a way to allow users to connect multiple accounts from their metamask wallet?

### DISCUSSION THREAD

**Discussion by Deveshb15** [wevm/wagmi#147] [Q&A]
Usually what I've seen is people do have more than 1 account in their metamask wallet, so is there a way for users to connect other wallet too directly from our website instead of changing accounts in the metamask wallet manually?

**Comment by tmm** [comment]
Not currently, but thinking about what this would look like.

Any specific behavior you would want or not want?

**Reply by Deveshb15** [reply]
Need to group and display transaction data of both the accounts so wanted a way to allow users to connect first wallet and then give a button to connect another wallet(+1 wallet) too just to get their wallet address and display data

**Reply by agustif** [reply]
Could you abstract away two different providers from wagmiProvider for each wallet. So each should live in its own react context? Not sure if this would work but it's what I'd try

**Reply by anoushk1234** [reply]
@tmm is this on the roadmap, i need to allow the user to connect another wallet from their metamask/rainbow

**Reply by agustif** [reply]
I ended up saving the wallet as -address- on the user in database, you could do the same and save as an array several addresses.

make the wallet owner sign a nonce to prove they're the owners.

List from all the accounts linked.

I liked how Skiff solved the UI. Mine is a bit more barebones. But it works

**Reply by Deveshb15** [reply]
did the same, saving the addresses as an array in db and making the user sign a nonce to prove the ownership

**Comment by victorpavlenko** [comment]
@tmm is this already work? Any way to request/swtich accounts from UI?

**Reply by tmm** [reply]
Likely coming in wagmi v2
